### PR TITLE
Add RAI checker for installed backends

### PR DESCRIPTION
**Pull request by al-rigazzi**
This PR mainly adds `CONFIG.get_installed_backends`, a property returning a list of installed Redis AI backends.

This is used e.g. in tests, to check whether the corresponding test should be run. This also helps distinguishing between tests we cannot run because there is no backend, and tests we cannot run because the library is not installed (e.g. `test_freeze_model` which does not need Redis AI).

Notice that for backend tests, we also check whether the module is (still) available, to prevent cases in which a user might have built RedisAI in another env.

### CODE REVIEW

**Comment by Spartee** [comment]
Thanks for catching this and taking care of it. My main thought on this is: if it's only used in tests... should it just be in ``conftest.py``? 

I suppose it might be nice to have this functionality internally. I can think of at least one use case which is to be able to discern wether a user has built any backends at all when initializing an ``Orchestrator``. Can you think of others I'm missing? 

If we do want it internally, it's not necessarily a configuration option as much as it is a helper function. ``smartsim/_core/util/helpers.py`` might be a better location. That is unless we want the user to be able to pass in the locations of the backends at some point... This is an option RedisAI supports with ``BACKENDSPATH``. I think if we do leave it in config, we should make sure that's the route we want to go and if so, make a note about it in the code.

I'm personally leaning towards a helper function. thoughts??

**Review by Spartee — COMMENTED** [review]
See comment above

**Comment by al-rigazzi** [comment]
@Spartee Yeah, I think we can move it to `smartsim/_core/util/helpers.py` and promote it to function (instead of it being a property of `CONFIG` as it is now).

File: tests/backends/test_tf.py [hunk 1]
```diff
@@ -4,23 +4,25 @@
...
 
-should_run = True
+tf_available = True
 try:
     from tensorflow import keras
 
     from smartsim.tf import freeze_model
 except (ImportError, SmartSimError):
-    should_run = False
+    tf_availavble = False
```
Comment (MattToast) [inline tests/backends/test_tf.py:17]: `tf_available`

File: smartsim/_core/config/config.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
 import os
 from functools import lru_cache
 from pathlib import Path
+from typing import List
```
Comment (MattToast) [inline smartsim/_core/config/config.py:30]: Remove unused typing module

**Review by MattToast — COMMENTED** [review]
Quick fix on unused import and typo-ed variable name that i think is causing tests to error. Otherwise lgtm!!

File: tests/backends/test_tf.py (hunk 1 above)
Comment (al-rigazzi) [reply tests/backends/test_tf.py]: Thanks, fixed

File: smartsim/_core/config/config.py (hunk 2 above)
Comment (al-rigazzi) [reply smartsim/_core/config/config.py]: Removed

**Review by Spartee — APPROVED** [review]
LGTM. Eventually we want to add this to the CLI. maybe a `smart build --check-installed`` command? Either way, glad to have this functionality.

**Pull request merged**